### DISCUSSION TITLE
Support dynamic content in chapters

### DISCUSSION THREAD

**Discussion by theDanielJLewis** [Podcastindex-org/podcast-namespace#511] [Ideas]
Talking with @MartinMouritzen today, we came up with an idea to help time-based features account for dynamic content.

## The problem

Imagine we support my proposal for cross-app comments (#505) and comments have a timestamp of how many seconds into the episode the comment was submitted (either when the comment button was pressed, or when the comment was submitted). A player could display those comments in some sort of timeline.

But if there's any dynamically inserted content before the comment, the timestamp would be off.

Additionally, dynamically inserted content messes with the transcript timing.

## The solution is in chapters

Our idea is to insert a chapter for any kind of dynamic content. Such a chapter would probably be best to not be included in the table of contents, and building on my proposal to extend chapter capabilities (#469), the chapter could have a "type" or other attribute that helps podcast apps process it.

The goal would be for the chapter to match the length of the dynamically inserted content, default to _not_ enable any kind of skipping or displaying of the chapter, but it would trigger the player to offset any following transcripts, comments, boostagrams, or even other chapters. (If the system is already inserting a chapter for dynamic content, I would expect it to appropriately change the timings of all following chapters, and then shift them back if the dynamic content is changed or removed.)

@tomrossi7, what does Buzzsprout do with chapters and transcripts when you dynamically insert any content?

**Comment by tomrossi7** [comment]
Buzzsprout always keeps the chapters and transcripts with the current episode audio. I think the spec should simply state that these files are accurate at the time of download. Does that make sense?

**Reply by theDanielJLewis** [reply]
That makes sense, but I foresee problems with apps looking for updated chapters during playback or when an episode is paused and resumed.

Thus, that reveals another problem. What if the fully downloaded file has no dynamically inserted content, but the chapters do? There would be no way to match the chapters with the file unless there could be some kind of hash to match between them.

**Reply by tomrossi7** [reply]
Just like the audio content, once it's downloaded, it's downloaded. If you wanted to update it, you would need to download it again. The audio, chapters, and transcript all must be downloaded at the same time.
